src/darwin/Framework/CHIP/MTRBaseDevice.h

```diff
@@ -227,17 +234,25 @@ extern NSString * const MTRArrayValueType;
                          completion:(MTRDeviceResponseHandler)completion MTR_NEWLY_AVAILABLE;
 
 /**
- * Subscribe an attribute in a designated attribute path
+ * Subscribes to the given attribute path on the device.
```

Comment:
So the spec has the concept of an "Attribute Path" which an include wildcards and a "Concrete Attribute Path" which does not.  Our MTRAttributePath is a "Concrete Attribute Path"...  And yes, maybe this should just be subscribeAttributes, good idea.  How about `subscribeToAttributes`?

> Is it legal to pass nil for all 3 parameters to subscribe to every attribute of a device?

For this specific API, yes, it is.

> What happens if you make multiple subscribe calls that overlap (e.g. due to wildcards in different places), do you get multiple subscriptions?

If they set the "keep previous subscriptions" bit in the subscribe params to true, yes.  Otherwise, multiple subscribes will lead to just the last one being around.

Reply:
`subscribeToAttributes` sounds good to me, not sure if there's other instances of similar methods we want to keep consistent with though.